File: Java Automation QA Sample/src/main/java/in/wrapstore/automation/utils/WaitUtil.java
package in.wrapstore.automation.utils;

import java.time.Duration;

import org.openqa.selenium.*;
import org.openqa.selenium.support.ui.*;

public class WaitUtil {

    public static void waitForVisibility(WebDriver driver, By locator, int timeout) {
        new WebDriverWait(driver, Duration.ofSeconds(timeout))
                .until(ExpectedConditions.visibilityOfElementLocated(locator));
    }

    public static void waitForClickability(WebDriver driver, By locator, int timeout) {
        new WebDriverWait(driver, Duration.ofSeconds(timeout))
                .until(ExpectedConditions.elementToBeClickable(locator));
    }
    public static void waitForTitleContains(WebDriver driver, String title, int timeout) {
        new WebDriverWait(driver, Duration.ofSeconds(timeout))
                .until(ExpectedConditions.titleContains(title));
    }

}

File: Python Automation QA Sample/utils/wait_util.py
# in/wrapstore/automation/utils/wait_util.py

from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class WaitUtil:
    """Utility class for explicit waits."""

    @staticmethod
    def wait_for_visibility(driver, locator: tuple, timeout: int):
        """Waits until the element located by the given locator is visible."""
        WebDriverWait(driver, timeout).until(
            EC.visibility_of_element_located(locator)
        )

    @staticmethod
    def wait_for_clickability(driver, locator: tuple, timeout: int):
        """Waits until the element is clickable."""
        WebDriverWait(driver, timeout).until(
            EC.element_to_be_clickable(locator)
        )

    @staticmethod
    def wait_for_title_contains(driver, title_substring: str, timeout: int):
        """Waits until the page title contains the given substring."""
        WebDriverWait(driver, timeout).until(
            EC.title_contains(title_substring)
        )
